test/serializers/serializer_for_test.rb

```diff
@@ -84,6 +89,11 @@ def test_serializer_inherited_serializer
           assert_equal ProfileSerializer, serializer
         end
 
+        def test_serializer_inherited_serializer_with_namespace
+          serializer = ActiveModel::Serializer.serializer_for(@my_profile, namespace: SerializerNamespace)
+          assert_equal SerializerNamespace::ProfileSerializer, serializer
+        end
```

Comment:
this fails without this change?

Reply:
Yes. The reason is that when looking up the serializer for superclass, in `#get_serializer_for` the namespace is not passed along in the method call.